backend/de.metas.manufacturing/src/main/java/de/metas/manufacturing/generatedcomponents/ComponentGeneratorRepository.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * #%L
+ * de.metas.manufacturing
+ * %%
+ * Copyright (C) 2020 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.manufacturing.generatedcomponents;
+
+import com.google.common.collect.ImmutableMap;
+import de.metas.javaclasses.JavaClassId;
+import de.metas.product.ProductId;
+import de.metas.util.GuavaCollectors;
+import de.metas.util.Services;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import org.adempiere.ad.dao.IQueryBL;
+import org.compiere.model.I_PP_ComponentGenerator;
+import org.compiere.model.I_PP_ComponentGenerator_Param;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+@Repository
+public class ComponentGeneratorRepository
+{
+	final IQueryBL queryBL = Services.get(IQueryBL.class);
+
+	// TODO tbp: CCache this
+
+	@Nullable
+	public PP_ComponentGenerator getByProductId(@NonNull final ProductId productId)
+	{
+		final I_PP_ComponentGenerator po = queryBL.createQueryBuilder(I_PP_ComponentGenerator.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_PP_ComponentGenerator.COLUMNNAME_M_Product_ID, productId)
+				.create()
+				.first();
+
+		if (po == null)
+		{
+			return null;
+		}
+
+		final List<I_PP_ComponentGenerator_Param> paramsPO = queryBL.createQueryBuilder(I_PP_ComponentGenerator_Param.class)
```

Comment:
when u will load all records in one round, keep in mind to avoid SQL N+1 here

Reply:
done